### PR TITLE
Feat/379 textarea CSS API

### PR DESCRIPTION
**Pull request by caseybaggz**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?
contributes #379 

## What is the new behavior?
Adds Textarea CSS api and sandbox example

### CODE REVIEW

File: packages/headless-styles/src/components/Textarea/textareaCSS.module.css [hunk 1]
```diff
@@ -0,0 +1,86 @@
+.textareaBase {
+  appearance: none;
+  background: var(--ps-neutral-surface-weak);
+  border-color: var(--ps-neutral-border);
+  border-image: initial;
+  border-radius: 6px;
+  border-style: solid;
+  border-width: 1px;
+  color: white;
```
Comment (Rykus0) [inline packages/headless-styles/src/components/Textarea/textareaCSS.module.css:9]: Text isn't readable in light mode.

File: packages/headless-styles/src/components/Textarea/shared.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    resize: options?.resize ?? defaultTextareaOptions.resize,
+    tech: options?.tech ?? defaultTextareaOptions.tech,
+    value: options?.value ?? defaultTextareaOptions.value,
+  }
+}
+
+export function createTextareaOptions(options: TextareaOptions) {
+  const { describedBy } = options
+  const a11yProps = createA11yProps(options)
+  const describedByProps = describedBy && {
```
Comment (Rykus0) [inline packages/headless-styles/src/components/Textarea/shared.ts:38]: I think we should combine this with `a11yProps` if we can

File: packages/headless-styles/tests/textarea/textareaCSS.test.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  test('should accept a resize type', () => {
+    expect(getTextareaProps({ ...options, resize: 'initial' })).toEqual(result)
+    expect(getTextareaProps({ ...options, resize: 'horizontal' })).toEqual({
+      ...result,
+      className: `${baseClass} horizontalTextarea`,
+    })
+    expect(getTextareaProps({ ...options, resize: 'vertical' })).toEqual({
+      ...result,
+      className: `${baseClass} verticalTextarea`,
+    })
```
Comment (Rykus0) [inline packages/headless-styles/tests/textarea/textareaCSS.test.ts:45]: Missing the 'none' case

File: packages/headless-styles/src/components/Textarea/textareaCSS.module.css (hunk 1 above)
Comment (caseybaggz) [reply packages/headless-styles/src/components/Textarea/textareaCSS.module.css:9]: Alas...finally having no white variable is an issue. 😭 Guess I'll just use whatever we are using on Input...not sure why it's different for TA anyways? 🤷‍♀️

File: packages/headless-styles/tests/textarea/textareaCSS.test.ts (hunk 3 above)
Comment (caseybaggz) [reply packages/headless-styles/tests/textarea/textareaCSS.test.ts:45]: Good eye!

File: packages/headless-styles/src/components/Textarea/shared.ts (hunk 2 above)
Comment (caseybaggz) [reply packages/headless-styles/src/components/Textarea/shared.ts:38]: @Rykus0 there is a reason we don't do that in `createA11yProps` which are that not all inputs/components using that function require the `describedBy` attr. So, adding that logic into the function just for a couple of use cases is by nature a bad functional programming and clean code practice. It's better to keep it separate so the function stays pure which will reduce the risk of bugs happening over the lifetime of the code.

**Review by Rykus0 — APPROVED** [review]

**Pull request merged**